File: frontend/henn-laszlo/projects/content-data-access/src/lib/models/admin-content-page-details.ts
import { ContentPageKey } from './content-page-key';

export interface AdminContentPageDetails {
  readonly id: string;
  readonly key: ContentPageKey;
  readonly titleHu: string;
  readonly titleEn: string | null;
  readonly contentHu: string;
  readonly contentEn: string | null;
  readonly isPublished: boolean;
  readonly updatedAtUtc: string;
}

File: backend/src/Modules/Modules.Content.Application/ContentPages/GetAdminByKey/AdminContentPageDetails.cs
namespace Modules.Content.Application.ContentPages.GetAdminByKey;

public sealed record AdminContentPageDetails(
    Guid Id,
    string Key,
    string TitleHu,
    string? TitleEn,
    string ContentHu,
    string? ContentEn,
    bool IsPublished,
    DateTime UpdatedAtUtc);